Content.Server/Verbs/VerbSystem.cs

```diff
@@ -24,157 +26,110 @@ public override void Initialize()
-        }
-
-        public void AddContainerVisibility(IPlayerSession session)
-        {
-            if (!_seesThroughContainers.Add(session))
-            {
-                return;
-            }
-
-            var message = new PlayerContainerVisibilityMessage(true);
-            RaiseNetworkEvent(message, session.ConnectedClient);
+            SeeAllContextPlayers.Clear();
         }
 
-        public void RemoveContainerVisibility(IPlayerSession session)
+        public void ToggleSeeAllContext(IPlayerSession player)
         {
-            if (!_seesThroughContainers.Remove(session))
+            if (!SeeAllContextPlayers.Add(player))
             {
-                return;
+                SeeAllContextPlayers.Remove(player);
             }
 
-            var message = new PlayerContainerVisibilityMessage(false);
-            RaiseNetworkEvent(message, session.ConnectedClient);
-        }
-
-        public bool HasContainerVisibility(IPlayerSession session)
-        {
-            return _seesThroughContainers.Contains(session);
+            SetSeeAllContextEvent args = new() { CanSeeAllContext = SeeAllContextPlayers.Contains(player) };
+            RaiseNetworkEvent(args, player.ConnectedClient);
         }
 
-        private void UseVerb(UseVerbMessage use, EntitySessionEventArgs eventArgs)
+        /// <summary>
+        ///     Called when asked over the network to run a given verb.
+        /// </summary>
+        public void HandleTryExecuteVerb(TryExecuteVerbEvent args, EntitySessionEventArgs eventArgs)
         {
-            if (!EntityManager.TryGetEntity(use.EntityUid, out var entity))
-            {
-                return;
-            }
-
             var session = eventArgs.SenderSession;
             var userEntity = session.AttachedEntity;
 
             if (userEntity == null)
             {
-                Logger.Warning($"{nameof(UseVerb)} called by player {session} with no attached entity.");
+                Logger.Warning($"{nameof(HandleTryExecuteVerb)} called by player {session} with no attached entity.");
                 return;
             }
 
-            foreach (var (component, verb) in VerbUtility.GetVerbs(entity))
+            if (!EntityManager.TryGetEntity(args.Target, out var targetEntity))
             {
-                if ($"{component.GetType()}:{verb.GetType()}" != use.VerbKey)
-                {
-                    continue;
-                }
-
-                if (!VerbUtility.VerbAccessChecks(userEntity, entity, verb))
-                {
-                    break;
-                }
-
-                verb.Activate(userEntity, component);
-                break;
+                return;
             }
 
-            foreach (var globalVerb in VerbUtility.GetGlobalVerbs(Assembly.GetExecutingAssembly()))
+            var verbs = GetVerbs(targetEntity, userEntity, args.Type)[args.Type];
+            var verb = verbs.Where((v) => v.Key == args.VerbKey).FirstOrDefault();
```

Comment:
iirc this kind of LINQ is really slow

Reply:
Yeah this really should have just been a for loop.